lib/ansible/config/routing.yml

```diff
@@ -0,0 +1,54 @@
+# these entries are consulted for all ansible.builtin lookups (using the same mechanism real collections do), and
+# is also a "last chance" consultation for unqualified plugins before failing
+plugin_routing:
+  connection:
+    formerly_core_winrm:
+      redirect: ansible.builtin.winrm
+  module_utils:
+    # FIXME: implement powershell/csharp module_utils support (ideally just subdir + extension support with FQPN target)
+    # FIXME: implement module_utils subdir support; subdir syntax is subdir1.subdir2.resource, subdir1/subdir2/resource, or?
+    formerly_core:
+      # TODO: allow FQCN or require FQPN for module_utils?
+      # TODO: special case these to assume we always want to allow direct import?
+      redirect: ansible_collections.testns.testcoll.plugins.module_utils.base
+  modules:
+    # TODO: is this a workable generic replacement for flatmapping and internal aliasing? Should be more performant, and less wonky, but not automatic like flatmapping...
+    # FIXME: implement partially-qualified subdir support for redirect
+    # eg: toplevel_module:
+    #       redirect: .subdir1.subdir2.toplevel_module
+    # NB: optional extension-specific redirects allow disambiguation of colliding plugins (eg setup.py/.ps1, slurp.py/.ps1)
+    setup.ps1:
+      redirect: testns.testcoll.win_selfcontained
+    formerly_core_ping:
+      redirect: testns.testcoll.ping  # testns.testcoll.module_subdir.subdir_ping_module
+    formerly_core_win_ping:
+      redirect: testns.testcoll.win_selfcontained
+  filter:
+    # NB, both sides of this redirect use the exposed filter name; we're ignoring the containing Python module
+    formerly_core_filter:
+      redirect: ansible.builtin.bool
+  lookup:
+    formerly_core_lookup:
+      redirect: testns.testcoll.mylookup
+  shell:
+    formerly_core_powershell:
+      redirect: ansible.builtin.powershell
+  test:
+    # NB, both sides of this redirect use the exposed test name; we're ignoring the containing Python module
+    formerly_core_test:
+      redirect: ansible.builtin.search
+
+# FIXME: implement consumption code for this with shertel's collectionizing of module_default groups
```

Comment:
we are already generating a separate file with this action_groups.yml  https://github.com/ansible-collection-migration/community.amazon/blob/master/meta/action_groups.yml

Reply:
Right- this is the builtin thing that would redirect unqualified references to the 2.9 group names to the ones you're creating in the various collections. There are a couple other things in this PR that it looks like @s-hertel will need before the action groups changeover PR can be fully completed, so either she'd have to rebase her PR on this and track changes, or wait until it's merged. I don't think we're in a super big hurry for that to be working, so I'd advocate for the latter approach, but whatevs. :)

Reply:
I don't have a strong opinion on sequencing or implementation here - but if this routing option exists and allows that we can express a mapping on an old module -> new module basis that's great.  Specifically I want to be able to make a list like
```
  os_server:
    redirect: openstack.cloud.server
```
and not need to keep the module named os_server inside of the `openstack.cloud` collection.

So just tell me where to put the mapping and I'm on it. :)

Reply:
@emonty  in meta/routing.yml, see https://github.com/ansible-collections/amazon.aws/blob/master/meta/routing.yml as existing example (thought those are all deprcations, redirect works in same way.)